packages/patternfly-3/react-tree-view/src/components/TreeView/index.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { Tree } from 'react-wooden-tree';
+
+require('react-wooden-tree/dist/react-wooden-tree.css');
+require('./style.css');
+
+const data = [
+  {
+    text: 'Parent 1',
+    icon: 'fa fa-folder',
+    nodes: [
+      {
+        text: 'Child 1',
+        nodes: [{ text: 'Grandchild 1' }, { text: 'Grandchild 2' }]
+      },
+      {
+        text: 'Child 2',
+        nodes: [{ text: 'Grandchild 1' }, { text: 'Grandchild 2' }]
+      }
+    ]
+  },
+  {
+    text: 'Parent 2',
+    icon: 'fa fa-folder',
+    nodes: [{ text: 'Child 1' }]
+  },
+  {
+    text: 'Parent 3',
+    icon: 'fa fa-folder'
+  },
+  {
+    text: 'Parent 4',
+    icon: 'fa fa-folder'
+  },
+  {
+    text: 'Parent 5',
+    icon: 'fa fa-folder'
+  }
+];
+
+// const lazyLoad(node) {
+//   const isWorking = true;
+//   return new Promise((resolve, reject) => {
+//     setTimeout(() => {
+//       if (isWorking) {
+//         resolve([]);
+//       } else {
+//         reject(new Error('Something happened.'));
+//       }
+//     }, 2000);
+//   });
+// }
+
+const actionMapper = {
+  'state.expanded': Tree.nodeExpanded,
+  'state.checked': Tree.nodeChecked,
+  'state.disabled': Tree.nodeDisabled,
+  'state.selected': Tree.nodeSelected,
+  nodes: Tree.nodeChildren,
+  loading: Tree.nodeLoading
+};
+
+class TreeView extends React.Component {
+  /**
+   * Constructor.
+   * @param {{}} props
+   */
+  constructor(props) {
+    super(props);
+
+    this.data = Tree.initTree(data);
+
+    this.state = {
+      tree: this.data
+    };
+
+    this.onDataChange = this.onDataChange.bind(this);
+    // this.lazyLoad = this.lazyLoad.bind(this);
+  }
+
+  /**
+   * The callback function for changing data in the tree.
+   *
+   * @param {string} nodeId The nodeId of the node.
+   * @param {string} type The field name which changed.
+   * @param {boolean} value The new value to assign.
+   */
+  onDataChange(nodeId, type, value) {
+    let node = Tree.nodeSelector(this.data, nodeId);
+    if (node == null) {
+      return;
+    }
+
+    if (actionMapper.hasOwnProperty(type)) {
+      node = actionMapper[type](node, value);
+      this.data = Tree.nodeUpdater(this.data, node);
+    } else {
+      // console.log(nodeId, type, value);
+    }
+
+    this.setState({ tree: this.data });
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <Tree
+          nodeIcon="fa fa-file-o"
+          data={this.state.tree}
+          onDataChange={this.onDataChange}
+          lazyLoad={this.lazyLoad}
```

Comment:
Should we spread props into `<Tree />` for the remaining react-wooden-tree options?

Reply:
I am sorry, but I don't understand the question: what do you mean under spread?

Reply:
I was thinking, say, if the consumer wants to take advantage of the `loadingIcon` option for `<Tree />`.  With the current implementation, it does not seem possible to take advantage of the configuration.

If we added
```js
<Tree
  ...
  {...this.props}
/>
```
it would allow the consuming application to make use of all the options available [on this list](https://www.npmjs.com/package/react-wooden-tree#tree-options)

Or maybe, we could explicitly declare propTypes for all of the options?

Reply:
So if I understood correctly, you want to get the default values of the props which are defined in the tree which is defined [here](https://github.com/brumik/react-wooden-tree/blob/master/src/components/Tree.tsx#L547) and [here](https://github.com/brumik/react-wooden-tree/blob/master/src/components/Node.tsx#L314)? So it can be used in the parent component of the tree?

I cannot test but I pretty sure that you can use [String: Any] variable to define the props:
```typescript
<Tree {...this.propsForTheTree} />
```
this should work if the propsForTheTree is defined correctly.

Reply:
Definitely agree that

```js
<Tree {...this.propsForTheTree} />
```

would work :)  However, please forgive me if I am missing something, but that would only work if the downstream application was doing something like

```js
import { Tree } from 'react-wooden-tree'
// ...
<Tree { ...this.propsForTheTree } />
```

I have to admit that I'm not exactly sure how our yarn workspace works exactly, but my understanding is that after this package is added, the consuming application would be doing something like

```
import { TreeView } from '@patternfly/react-tree-view`;
// ...
<TreeView { ...this.propsForTheTree } />
```

in which case, spreading props into `TreeView` would not work.  What do you think?